src/Store/ExtensionClientStoreConnector.re

```diff
@@ -181,100 +240,95 @@ let start = (extensions, setup: Core.Setup.t) => {
       }
     );
 
-  let suggestionItemToCompletionItem:
-    Protocol.SuggestionItem.t => Model.Actions.completionItem =
-    suggestion => {
-      let completionKind =
-        suggestion.kind |> Option.bind(CompletionItemKind.ofInt);
-
-      {
-        completionLabel: suggestion.label,
-        completionKind,
-        completionDetail: suggestion.detail,
-      };
-    };
-
-  let suggestionsToCompletionItems = (suggestions: Protocol.Suggestions.t) =>
-    List.map(suggestionItemToCompletionItem, suggestions);
-
   let getAndDispatchCompletions =
-      (~languageFeatures, ~fileType, ~uri, ~completionMeet, ~position, ()) => {
-    let providers =
-      LanguageFeatures.getSuggestProviders(fileType, languageFeatures);
-
-    providers
-    |> List.iter((provider: LanguageFeatures.SuggestProvider.t) => {
-         Log.infof(m =>
-           m(
-             "Completions - getting completions for suggest provider: %n",
-             provider.id,
-           )
-         );
-         let completionPromise: Lwt.t(option(Protocol.Suggestions.t)) =
-           ExtHostClient.getCompletions(
-             provider.id,
-             uri,
-             position,
-             extHostClient,
-           );
+      (~languageFeatures, ~buffer, ~meet, ~position, ()) => {
+    let completionPromise =
+      Model.LanguageFeatures.requestCompletions(
+        ~buffer,
+        ~meet,
+        ~position,
+        languageFeatures,
+      );
 
-         let _ =
-           Lwt.bind(
-             completionPromise,
-             completions => {
-               switch (completions) {
-               | None => Log.info("No completions for provider")
-               | Some(completions) =>
-                 let completionItems =
-                   suggestionsToCompletionItems(completions);
-                 dispatch(
-                   Model.Actions.CompletionAddItems(
-                     completionMeet,
-                     completionItems,
-                   ),
-                 );
-               };
-               Lwt.return();
-             },
+    Lwt.bind(
+      completionPromise,
+      completions => {
+        dispatch(Model.Actions.CompletionAddItems(meet, completions));
+        Lwt.return();
+      },
+    );
```

Comment:
Does this need to return a promise? I see it's just discarded later anyway, so seems like this could use `Lwt.on_success` instead.

Reply:
Good catch